### PR TITLE
Move ScyllaDB main config file generation from sidecar to operator

### LINKED ISSUE #1714: Move scylladb config creation from sidecar to the operator
ScyllaDB config should be created in the operator where all the data and informers are available without taking extra cpu or requiring binary copies (sidecar).

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR moves config generation from sidecar to the operator. This will allow us to move more flags into the config as well in the future and limit the sidecar scope, hopefully to zero one day.

Note: The Alternator work builds on top of this PR.

**Which issue is resolved by this Pull Request:**
Fixes #1714

First step to #1684 and #604

### CODE REVIEW

File: pkg/controller/scyllacluster/sync_config.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+) ([]metav1.Condition, error) {
+	var errs []error
+	var progressingConditions []metav1.Condition
+
+	cm, err := MakeManagedScyllaDBConfig(sc)
+	if err != nil {
+		return progressingConditions, fmt.Errorf("can't make managed scylladb config: %w", err)
+	}
+
+	cm.SetOwnerReferences([]metav1.OwnerReference{
```
Comment (zimnx) [inline pkg/controller/scyllacluster/sync_config.go:28]: why this is outside of `MakeManagedScyllaDBConfig`?

File: pkg/sidecar/config/config.go [hunk 2]
```diff
@@ -34,6 +30,7 @@ const (
 	configDirScyllaD                    = "/etc/scylla.d"
 	scyllaYAMLPath                      = configDirScylla + "/" + naming.ScyllaConfigName
 	scyllaYAMLConfigMapPath             = naming.ScyllaConfigDirName + "/" + naming.ScyllaConfigName
+	scyllaYAMLManagedConfigMapPath      = naming.ScyllaDBManagedConfigDir + "/scylladb-managed-config.yaml"
```
Comment (zimnx) [inline pkg/sidecar/config/config.go:33]: `scylladb-managed-config.yaml` could be const, and then synchronized between this variable and actual ConfigMap content.

File: pkg/controller/scyllacluster/resource.go [hunk 3]
```diff
@@ -396,6 +399,17 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
...
 								},
 							},
+							{
+								Name: "scylladb-managed-config",
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: naming.GetScyllaDBManagedConfigCMName(c.Name),
+										},
+										Optional: ptr.To(false),
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply pkg/controller/scyllacluster/resource.go:409]: it wasn't available before but I don't mind changing that

File: pkg/controller/scyllacluster/sync_config.go (hunk 1 above)
Comment (tnozicka) [reply pkg/controller/scyllacluster/sync_config.go:28]: modeled after https://github.com/scylladb/scylla-operator/blob/ef4406fd838ed8db3eec6dd87786218ee265b88a/pkg/controller/scylladbmonitoring/sync_grafana.go#L477-L486

the idea behind that is that whoever creates it sets it, so possibly other controllers can use it too. I admit it's not that likely, so I can see this going both ways a bit.

File: pkg/sidecar/config/config.go (hunk 2 above)
Comment (tnozicka) [reply pkg/sidecar/config/config.go]: done

File: pkg/controller/scyllacluster/resource.go [hunk 4]
```diff
@@ -272,6 +274,7 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
 	}
 	rackTemplateAnnotations[naming.PrometheusScrapeAnnotation] = naming.LabelValueTrue
 	rackTemplateAnnotations[naming.PrometheusPortAnnotation] = "9180"
+	rackTemplateAnnotations["inputs-hash"] = inputsHash
```
[the remark this answers is not in the document]
Comment (tnozicka) [reply pkg/controller/scyllacluster/resource.go]: done

File: pkg/controller/scyllacluster/sync_config.go (hunk 1 above)
Comment (zimnx) [reply pkg/controller/scyllacluster/sync_config.go:28]: I would rather keep it consistent at least across CRD. Having object construction split into multiple places is error prone. 
We can extract it out, once we actually need it.

**Comment by zimnx** [comment]
PR title suggest entire config generation is moved to Operator, but only subset is. Lets fix that.

**Comment by tnozicka** [comment]
> PR title suggest entire config generation is moved to Operator, but only subset is. Lets fix that.

Which ones aren't? There are command line flags but those are not config. If you had these in mind should I reword "config generation" as "config file generation"?

**Comment by zimnx** [comment]
>Which ones aren't? 
Snitch config `cassandra-rackdc.properties` is not moved.

File: pkg/controller/scyllacluster/sync_config.go (hunk 1 above)
Comment (tnozicka) [reply pkg/controller/scyllacluster/sync_config.go]: done

**Comment by tnozicka** [comment]
Renamed + created #1714 to track it, thx.

**Review by zimnx — APPROVED** [review]
Thanks!
/lgtm

**Comment by tnozicka** [comment]
https://github.com/scylladb/scylla-operator/issues/1718#issuecomment-1935835282
/retest

**Pull request merged**